### PR TITLE
Switch on javascript support

### PR DESCRIPTION
**Pull request by airblade**
This PR activates the plugin for javascript files.

I wasn't sure what to do about signature completions: the [code](https://github.com/kiteco/vim-plugin/blob/ffb3500a5b018e6469385ec4b9080b392518f210/autoload/kite/signature.vim) digs out values from nested keys with `python` in the name (just search for "python" on that page).

### CODE REVIEW

File: autoload/kite/languages.vim [hunk 1]
```diff
@@ -10,6 +10,10 @@ function! kite#languages#supported_by_plugin()
     return 1
   endif
 
+  if &filetype == 'javascript' && index(g:kite_supported_languages, 'javascript') != -1
```
Comment (jansorg) [inline autoload/kite/languages.vim:13]: Which file extensions map to vim's javascript?
Kite supports *.js, and *.jsx.  It also support *.vue, which is related to JavaScript.

File: plugin/kite.vim [hunk 2]
```diff
@@ -22,7 +22,7 @@ filetype on
 
 
 if !exists('g:kite_supported_languages')
-  let g:kite_supported_languages = ['python']
+  let g:kite_supported_languages = ['python', 'javascript']
```
Comment (jansorg) [inline plugin/kite.vim:25]: Ideally the plugin would query Kite's enabled languages? Is this already implemented in the plugin?
Python is always enabled, but JS or Golang could still be turned off at this time.

**Comment by jansorg** [comment]
Please add a feature flag to let a user control the JS support. It should be a setting and off by default, i.e. JS support disabled.

**Comment by airblade** [comment]
The flag is the variable `g:kite_supported_languages`.

**Comment by airblade** [comment]
Rebased and merged in 3c90428...7fa7ac4.

**Pull request closed without merging**